index.js

```
@@ -4,26 +4,40 @@ var vt = require('vector-tile');
 var request = require('request');
 var Protobuf = require('pbf');
 var format = require('util').format;
+var fs = require('fs');
 
 module.exports = function(args, callback) {
-    var zxy = args.url.match(/\/(\d+)\/(\d+)\/(\d+)/);
 
-    args.x = args.x === undefined ? zxy[2] : args.x;
-    args.y = args.y === undefined ? zxy[3] : args.y;
-    args.z = args.z === undefined ? zxy[1] : args.z;
-
-    if (args.x === undefined || args.y === undefined || args.z === undefined) {
-        return callback(new Error(format("Could not determine tile z, x, and y from %s; specify manually with -z <z> -x <x> -y <y>", JSON.stringify(url))));
+    try {
+        var file = fs.lstatSync(args.uri);
+        if (file.isFile()) {
+            fs.readFile(args.uri, function(err, data) {
+                if (err) throw err;
+                readTile(data);
+            });
+        }
+    } catch(err) {
+        request.get({
+            url: args.uri,
+            gzip: true,
+            encoding: null
+        }, function (err, response, body) {
+            if (err) throw err;
+            readTile(body);
+        });
     }
```

👍  to @springmeyer's suggestion. This avoid using an exception for flow control, which should generally be avoided (one reason is it tends to hide legitimate errors -- for example imagine if `fs.lstatSync` throws a file permissions exception).